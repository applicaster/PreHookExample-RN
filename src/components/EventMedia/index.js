import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Dimensions,
  ImageBackground,
  StyleSheet,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

const screenWidth = Dimensions.get('window').width;
const styles = StyleSheet.create({
  image: {
    height: screenWidth,
    width: screenWidth,
  },
  headerVisor: {
    height: 130,
  },
});

class EventMedia extends Component {
  render() {
    const { imageUrl, width, height } = this.props;
    let imageStyles = styles.image;
    if (width !== height) {
      const aspectRatio = (width / height);
      imageStyles = {
        height: (screenWidth / aspectRatio),
        width: screenWidth,
      };
    }

    return (
      <ImageBackground
        style={ imageStyles }
        source={{ uri: imageUrl }}
      >
        <LinearGradient style={ styles.headerVisor } colors={['rgba(0,0,0,0.7)', 'transparent']} />
      </ImageBackground>
    );
  }
}

EventMedia.propTypes = {
  imageUrl: PropTypes.string,
  height: PropTypes.number,
  width: PropTypes.number,
};

export default EventMedia;
