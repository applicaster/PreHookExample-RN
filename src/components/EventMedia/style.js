import {
  Dimensions,
  StyleSheet,
} from 'react-native';

const screenWidth = Dimensions.get('window').width;
export const styles = StyleSheet.create({
  mediaItem: {
    height: screenWidth,
    width: screenWidth,
    backgroundColor: 'black',
  },
  videoItem: {
    position: 'absolute',
    top: 0,
    left: 0,
  },
  headerVisor: {
    height: 130,
  },
});
