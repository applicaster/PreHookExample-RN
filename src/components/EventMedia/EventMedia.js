import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ImageBackground, TouchableWithoutFeedback } from 'react-native';
import LinearGradient from '@applicaster/react-native-linear-gradient';
import { sendAnalyticEvent } from 'react-native-zapp-bridge';
import { IMAGE_DETAIL_CLICKED } from '../../constants/analyticEvents';
import { styles } from './style';
import EventVideo from '../EventVideo';
import { getMediaDimensions } from '../../utils/size';
import { SCREEN_MARGIN } from '../../constants/measurements';

class EventMedia extends Component {
  constructor(props) {
    super(props);
    this.showMediaDetails = this.showMediaDetails.bind(this);
  }
  
  showMediaDetails() {
    const { toggleModal, setActiveEventId, eventId, isInteractive } = this.props;
    if (!isInteractive) return;
    
    setActiveEventId(eventId);
    toggleModal({ modal: 'MediaModal' });
    sendAnalyticEvent(IMAGE_DETAIL_CLICKED, { eventId }).then().catch();
  }

  renderVideo() {
    const { eventId, width, height, videoUrl } = this.props;
    return <EventVideo eventId={eventId} videoUrl={videoUrl} width={width} height={height} />;
  }

  render() {
    const { isInteractive, imageUrl, height, width, videoUrl } = this.props;
    const mediaItem = (
      <ImageBackground
        style={getMediaDimensions({ height, width, screenMargin: SCREEN_MARGIN })}
        source={{ uri: imageUrl }}
      >
        {videoUrl && this.renderVideo()}
        <LinearGradient style={ styles.headerVisor } colors={['rgba(0,0,0,0.7)', 'transparent']} />
      </ImageBackground>);

    if (isInteractive) {
      return <TouchableWithoutFeedback onPress={this.showMediaDetails}>{mediaItem}</TouchableWithoutFeedback>;
    }

    return mediaItem;
  }
}

EventMedia.propTypes = {
  eventId: PropTypes.string,
  imageUrl: PropTypes.string,
  isInteractive: PropTypes.bool,
  height: PropTypes.number,
  videoUrl: PropTypes.string,
  width: PropTypes.number,
  toggleModal: PropTypes.func,
  setActiveEventId: PropTypes.func,
};

EventMedia.contextTypes = {
  navigation: PropTypes.object,
};

export default EventMedia;
