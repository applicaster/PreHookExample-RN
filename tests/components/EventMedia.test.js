import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EventMedia from '../../src/components/EventMedia/EventMedia';

const eventMediaProps = (height, width, videoUrl) => ({
  imageUrl: 'someurl',
  height,
  width,
  videoUrl,
  showMediaDetailsModal: () => {},
});

jest.mock('react-native-video', () => 'Video');

describe('EventMedia', () => {
  test('component renders correctly for squared images', () => {
    const html = renderer.create(
      <EventMedia {...eventMediaProps(300, 300)} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for squared images', () => {
    const html = renderer.create(
      <EventMedia {...eventMediaProps(300, 300)} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for media items with video', () => {
    const html = renderer.create(
      <EventMedia {...eventMediaProps(300, 100, 'someurl')} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
