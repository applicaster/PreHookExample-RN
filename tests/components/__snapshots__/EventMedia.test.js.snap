// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventMedia component renders correctly for media items with video 1`] = `
<View
  style={
    Object {
      "height": 2160,
      "width": 720,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "someurl",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 2160,
          "width": 720,
        },
        undefined,
      ]
    }
  />
  <EventVideo
    eventId={undefined}
    height={300}
    videoUrl="someurl"
    width={100}
  />
  <BVLinearGradient
    colors={
      Array [
        3003121664,
        0,
      ]
    }
    end={undefined}
    locations={null}
    start={undefined}
    style={
      Object {
        "height": 130,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 720,
        "zIndex": 2,
      }
    }
  />
</View>
`;

exports[`EventMedia component renders correctly for squared images 1`] = `
<View
  style={
    Object {
      "height": 720,
      "width": 720,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "someurl",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 720,
          "width": 720,
        },
        undefined,
      ]
    }
  />
  <BVLinearGradient
    colors={
      Array [
        3003121664,
        0,
      ]
    }
    end={undefined}
    locations={null}
    start={undefined}
    style={
      Object {
        "height": 130,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 720,
        "zIndex": 2,
      }
    }
  />
</View>
`;

exports[`EventMedia component renders correctly for squared images 2`] = `
<View
  style={
    Object {
      "height": 720,
      "width": 720,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "someurl",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 720,
          "width": 720,
        },
        undefined,
      ]
    }
  />
  <BVLinearGradient
    colors={
      Array [
        3003121664,
        0,
      ]
    }
    end={undefined}
    locations={null}
    start={undefined}
    style={
      Object {
        "height": 130,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 720,
        "zIndex": 2,
      }
    }
  />
</View>
`;
